.github/workflows/lint.yml

```diff
@@ -0,0 +1,27 @@
+name: Pylint
+
+on: [ push ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.6", "3.7", "3.8", "3.9", "3.10" ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Create virtual environment
+        run: |
+          python -m venv venv
+          . venv/bin/activate
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e .[dev]
+      - name: Analysing the code with pylint
+        run: |
+          bash scripts/lint.sh
```

Comment:
I don't think it makes sense to run the linter for every Python version. It should be version agnostic as far as I know.

Reply:
Why using a script to run a single line when you just could do without it? Am I missing something=

Reply:
@runfalk The idea is that we want to make sure that the syntax of the file under lint is valid for all supported version of Python. For example, if we added a fancy new match case statement, as introduced in Python 3.10, the earlier Python version lint executions would fail with a message like `E0001: invalid syntax (<unknown>, line 2) (syntax-error)`. So this gives us the added benefit of automatically checking that our syntax is compliant with all supported versions of Python.

Now, maybe it would be enough to run the linter for Python 3.6 since that is our earliest supported version. But we don't want to only check for syntax that is too new, but also too **old** since things do get deprecated and removed between versions.

@kmpm In this particular case, I agree with you. However, I think that this is a good general pattern since it makes running the linter easier from command line (just `cd` to the project root and do `bash scripts/lint.sh`). We may want to add a `test.sh` later that runs unit tests. Also, I usually run `pylint` and `flake8` together for example in order to catch more potential errors, and in this case it makes more sense to have a script that runs both.